gpii/node_modules/lifecycleManager/src/PrivateMatchMaker.js

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Private matchMaker
+ *
+ * Copyright 2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+ "use strict";
+
+ var fluid = fluid || require("infusion"),
+     gpii = fluid.registerNamespace("gpii");
+
+ fluid.defaults("gpii.lifecycleManager.privateMatchMaker", {
+     gradeNames: ["fluid.component", "gpii.flowManager.matchMaking"],
+     invokers: {
+         doMatch: {
+             funcName: "gpii.lifecycleManager.privateMatchMaker.doMatch",
+             args: [
+                 "{that}",
+                 "{flowManager}.deviceReporterDataSource",
+                 "{arguments}.0",  // gpiiKey
+                 "{arguments}.1"   // preferences
+             ]
+         }
+     },
+     events: {
+         onSuccess: null,
+         onError: null
+     },
+     listeners: {
+         "onGpiiKey.getPreferences": "fluid.identity",
+         "onMatchDone.matchToSettings": {
+             listener: "gpii.flowManager.cloudBased.matchToSettings",
+             args: ["{arguments}.0", "{that}.events.onSuccess", "from privateMatchMaker"]
+         }
+     }
+ });
+
+ gpii.lifecycleManager.privateMatchMaker.doMatch = function (that, deviceReporterDataSource, gpiiKey, preferences) {
+     that.events.onGpiiKey.fire(gpiiKey);
+     that.events.onPreferences.fire(preferences);
+
+     var devicePromise = deviceReporterDataSource.get();
+     devicePromise.then(function (deviceData) {
+         fluid.log("Private Match Maker, doMatch - received device data: ", deviceData);
+         that.events.onDeviceContext.fire(deviceData);
+     }, function (err) {
+         fluid.log("Private Match Maker, doMatch - error at receiving device data: " + err.message);
+         that.events.onError.fire(err);
+     });
+
+     var requestPromise = fluid.promise();
+
+     that.events.onSuccess.addListener(function (payload) {
+         requestPromise.resolve(payload);
+         // To work around the issue that a fixed listener always binds to the first request prmoise
```

Comment:
Add a TODO that the proper solution to this is to request-scope the private matchMaker. It looks like in this case we will need to add a new kind of request, perhaps a "UserUpdate" request type, to go along with the UserLogon request types.